### PR TITLE
feat(apps): Include app job triggers in activity log

### PR DESCRIPTION
**Pull request by macobo**
## Problem

First, job triggers were logged but never actually displayed in plugins activity logs

Second, I want to start building towards [exposing app statistics to end users](https://github.com/PostHog/posthog/issues/12009). Having a list of past exports is useful for that and we can leverage existing data models.

Third, eventually exposing when the export finishes in the activity log improves the usefulness of this log. This however requires a bit of schema refactoring that will happen in a subsequent PR.

## Changes

- Add a new `trigger` type detail to activityLog for when an asynchronous action was triggered.
- Add (unused) `$job_id` parameter to job payloads - this will come in handy in a subsequent PR
- Expose job_triggered events in activity log

## How did you test this code?

Screenshots:

image

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
Looks good 

I like the explicit trigger 👍 

tests in here https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/ActivityLog/activityLogLogic.plugin.test.tsx have been useful for avoiding regression before now. Could be good to add one proving trigger is printed as expected

**Comment by macobo** [comment]
Since I'm dealing with stacked PRs, updating the tests in a following one. https://github.com/PostHog/posthog/pull/12068

File: posthog/api/plugin.py [hunk 1]
```diff
@@ -607,12 +610,15 @@ def job(self, request: request.Request, **kwargs):
         log_activity(
             organization_id=self.team.organization.id,
             # Users in an org but not yet in a team can technically manage plugins via the API
-            team_id=self.team.id,
+            team_id=self.team.pk,
```
Comment (tiina303) [inline posthog/api/plugin.py:613]: why this change?

File: posthog/api/plugin.py [hunk 2]
```diff
@@ -607,12 +610,15 @@ def job(self, request: request.Request, **kwargs):
...
             user=request.user,  # type: ignore
             item_id=plugin_config_id,
             scope="PluginConfig",  # use the type plugin so we can also provide unified history
             activity="job_triggered",
-            detail=Detail(name=self.get_object().plugin.name, changes=[Change(type="PluginConfig", action=job_type)]),
+            detail=Detail(
+                name=self.get_object().plugin.name,
+                trigger=Trigger(job_type=job_type, job_id=job_id, payload=job_payload),
```
Comment (tiina303) [inline posthog/api/plugin.py:620]: Should there be a unittest covering this?

**Review by tiina303 — APPROVED** [review]

File: posthog/api/plugin.py (hunk 1 above)
Comment (macobo) [reply posthog/api/plugin.py:613]: pk == id for this model, however we prefer `pk` throughout the codebase.
